lightning/src/blinded_path/payment.rs

```diff
@@ -260,6 +264,8 @@ pub struct ReceiveTlvs {
 	pub payment_constraints: PaymentConstraints,
 	/// Context for the receiver of this payment.
 	pub payment_context: PaymentContext,
+	/// An HMAC of `payment_context` along with a nonce used to construct it.
+	pub authentication: (Hmac<Sha256>, Nonce),
```

Comment:
So still write as we are for `BlindedPaymentTlvs` but read it as a separate component of `BlindedPaymentTlvs::Receive`? We'd need to pass it through to `Router::create_blinded_payment_paths`, `BlindedPaymentPath::new`, etc. Though, I think that would mean we wouldn't need to modify `Bolt12RefundContext` since the secret would be included.

Reply:
Gotcha, yeah that would reduce the amount of code we'd need to touch.

Reply:
Updated to use an `Option`, which is `take`n at verification time.